rapidast.py

```
@@ -404,18 +403,38 @@ def run():
             config.get("config.defectDojo.ssl", default=True),
         )
 
+    # Check GCS export configuration
+    gcs_exporter = None
+    if config.get("config.googleCloudStorage.bucketName"):
+        gcs_exporter = GoogleCloudStorage(
+            bucket_name=config.get("config.googleCloudStorage.bucketName"),
+            app_name=config.get_official_app_name(),
+            directory=config.get("config.googleCloudStorage.directory", None),
+            keyfile=config.get("config.googleCloudStorage.keyFile", None),
+        )
+
     # Run all scanners
     scan_error_count = 0
     for name in config.get("scanners"):
         logging.info(f"Next scanner: '{name}'")
 
-        ret = run_scanner(name, config, args, scan_exporter)
+        ret = run_scanner(name, config, args, dedo_exporter)
         if ret == 1:
             logging.info(f"scanner: '{name}' failed")
             scan_error_count = scan_error_count + 1
         else:
             logging.info(f"scanner: '{name}' completed successfully")
 
+    # Export all the scan results to GCS
+    # Note: This is done after all scanners have run,
+    #   unlike the DefectDojo export which needs to be done at the individual scanner level.
+    if gcs_exporter:
+        ret = gcs_exporter.export_scan(full_result_dir_path)
+        if ret == 1:
```

Instead of returning 0 or 1 to indicate success or failure, it would be cleaner to use exceptions, which aligns better with Python’s style. There are situations where the export could fail and it doesn't return 1, such as during tar creation or when copying files to GCP. These failures could be handled by catching exceptions here.

For example, you can catch the exception where the method is called
```
try:
    gcs_exporter.export_scan(full_result_dir_path)
    logging.info("Export to Google Cloud Storage completed successfully")
except Exception as e:
    logging.error("Export to Google Cloud Storage failed: %s", e)
```
I get that this approach is in line with how things are done in most of the RapiDAST code, but it could cause some exceptions to be missed, which might lead to poor user experience.

---

Thanks for your thoughtful comments. Updated.